src/cashweb/relay/index.ts

```diff
@@ -1010,6 +1038,19 @@ export class RelayClient extends ReadOnlyRelayClient {
       // If address data doesn't exist then add it
       const kind = entry.getKind()
 
+      if (kind === 'forward') {
```

Comment:
This stuff needs to be factored out to be able to call itself on the forwarded messages. There will be some goofiness here with stealth and p2pkh payments which you don't want forwarded or handled from a forwarded message.